data-prepper-plugins/common/src/test/java/org/opensearch/dataprepper/plugins/source/file/FileSourceTests.java

```diff
@@ -290,21 +321,18 @@ void setUp() {
 
         @Test
         void start_will_parse_codec_with_correct_inputStream() throws IOException {
-            createObjectUnderTest().start(buffer);
+            final FileInputStream decompressedStream = new FileInputStream(TEST_FILE_PATH_PLAIN);
```

Comment:
Rather than changing the whole test, could you just mock the `DecompressionEngine` to return exactly the bytes given?

Perhaps an alternative is to keep both tests.

Reply:
I cannot keep the test as it is: the test relies on the previous implementation's behavior: it tries to read from the input stream created in `FileSource.CodecFileStrategy#start` but because this stream is created in a try-with-resource (see [FileSource.java L181](https://github.com/opensearch-project/data-prepper/pull/5255/files/4684124bbb14fd6d01e4933bebc53f58d8d58d67#diff-79b9db4732930a1271f9653a6dbf5141c204951d567927290333645b8212b926R181)), it is closed when the test tries to read it again on [FileSourceTest.java L303](https://github.com/opensearch-project/data-prepper/pull/5255/files/4684124bbb14fd6d01e4933bebc53f58d8d58d67#diff-614c993d9d093322214d71b14e196d14069d6a778280c1c08f93145f4a1a23c4L303).
In other words, the test used to work because the input stream *was never properly closed*. I could reintroduce what I believe is a bug to keep the test as-is, but IMHO it would make more sense to validate the test is equivalent to the previous implementation.